x-pack/plugins/apm/public/components/shared/Links/rison_helpers.ts

```diff
@@ -52,40 +38,29 @@ function createG(query: RisonAPMQueryParams) {
   return g;
 }
 
-export function getRisonHref({
-  location,
-  pathname,
-  hash,
-  query = {}
-}: RisonHrefArgs) {
-  const currentQuery = toQuery(location.search);
+export function getRisonString(
```

Comment:
I think `getRisonString` has the potential to become the same centralized problem that we are trying to solve by decentralizing the links.

Different plugins uses rison encoding for different things. I don't think we can have one method for handling all logic for all plugins. Instead each link handler should handle it's own (rison) encoding.
If there are similarities we can maybe re-use them - eg. `getDateAsRison` or something like it.

Reply:
This is how I see it:
```js
function getTimepickerAsRison({rangeFrom, rangeTo, refreshPaused, refreshInterval}) {
  return rison.encode({...})
}
```

Btw. I'm not even sure if we want to call `rison.encode`  in case we need to merge it with other data. Might just want to return an object, and then let each link handler to the encoding.

Reply:
So probably more something like:
```js
function getTimepickerAsRison({
  rangeFrom,
  rangeTo,
  refreshPaused,
  refreshInterval
}) {
  return {
    _g: {
      'time.from': rangeFrom,
      'time.to': rangeTo,
      'refreshInterval.pause': refreshPaused,
      'refreshInterval.value': refreshInterval
    }
  };
}
```

Reply:
If you get the arguments from `urlParams` instead of `location` they will already have the defaults, and we don't have to handle that again here.

That poses a new question: perhaps we should have a `useUrlParams` hook, like the `useLocation` hook you already made? I think that would be very useful many places.